src/sentry/static/sentry/app/views/projectDashboard.jsx

```diff
@@ -36,12 +36,18 @@ const ProjectDashboard = createReactClass({
     return {
       statsPeriod: this.props.defaultStatsPeriod,
       activeEnvironment: null,
+      hasEnvironmentsFeature: new Set(this.context.organization.features).has(
+        'environments'
+      ),
       ...this.getQueryStringState(),
     };
   },
 
   componentWillMount() {
     this.props.setProjectNavSection('dashboard');
+
+    // Manually fire onLatestContextChange
+    this.onLatestContextChange(LatestContextStore.getInitialState());
```

Comment:
shouldn't this go in getInitialState though?

Reply:
It could, but can't fire `setState` in there, so would prefer to keep it here

Reply:
Wouldn't `getInitialState().activeEnvironment` == `(hasEnvironmentsFeature) && LatestContextStore.getInitialState().environment`